llvm/tools/llvm-profgen/ProfiledBinary.cpp

```diff
@@ -56,6 +56,10 @@ static cl::list<std::string> DisassembleFunctions(
     cl::desc("List of functions to print disassembly for. Accept demangled "
              "names only. Only work with show-disassembly-only"));
 
+static cl::opt<bool>
+    KernelBinary("kernel",
```

Comment:
Given we assume kernel binary has specific names, do we need an extra flag to tell the tool to generate profile for kernel? 

If we have this flag, what happens if the flag is used but binary provided doesn't match the predefined kernel binary name? Should we error out?

Reply:
I think this interface is easier than parsing the binary name. Usually the kernel image name is vmlinux.*. But there is a chance that an user image uses the same name. Or someone changes the kernel image name to whatever. I want users to confirm this is kernel perf data file by using the option.

If the user passes a wrong binary image, most like they won't get a profile because of the symbol and address mismatch.

Compared to AutoFDO's create_llvm_prof, llvm-profgen does not check the buildid. But that's separated issue. This patch is just the basic version that provides the kernel perf support.

Reply:
Ok, then this should be a static function?

Reply:
> If the user passes a wrong binary image, most like they won't get a profile because of the symbol and address mismatch.

Maybe useful to error out if kernel is requested but binary is not kernel?

Reply:
That actually what I want to avoid -- I don't want to assume a fixed name for the kernel. For my kernel builds, I don't change anything and the kernel image is vmlinux.*. But I don't know if others will rename the image.

Reply:
What I meant is error out when kernel image isn't found in mmap (we're already assuming names there). But no big deal either way.

Reply:
Done.